src/components/PortalWrapper/PortalWrapper.js

```diff
@@ -1,49 +1,65 @@
-import React, { PureComponent as Component } from 'react'
-import PropTypes from 'prop-types'
+import React from 'react'
 import ReactDOM from 'react-dom'
 import getComponentDefaultProp from '@helpscout/react-utils/dist/getComponentDefaultProp'
 import hoistNonReactStatics from '@helpscout/react-utils/dist/hoistNonReactStatics'
+import getComponentName from '@helpscout/react-utils/dist/getComponentName'
 import Animate from '../Animate'
 import KeypressListener from '../KeypressListener'
-import { default as Portal, propTypes as portalTypes } from '../Portal'
+import Portal, { propTypes } from '../Portal'
 import Keys from '../../constants/Keys'
 import {
   createUniqueIDFactory,
   createUniqueIndexFactory,
 } from '../../utilities/id'
 import { setupManager } from '../../utilities/globalManager'
 import { classNames } from '../../utilities/classNames'
-import { requestAnimationFrame } from '../../utilities/other'
+import { isFunction } from '../../utilities/is'
+import { noop, requestAnimationFrame } from '../../utilities/other'
 import matchPath from '../../utilities/react-router/matchPath'
 import Content from './Content'
 
 const defaultOptions = {
   id: 'PortalWrapper',
+  timeout: 100,
 }
 
 const managerNamespace = 'BluePortalWrapperGlobalManager'
 
 const PortalWrapper = (options = defaultOptions) => ComposedComponent => {
-  const propTypes = portalTypes
-
-  const defaultProps = {
-    isOpen: false,
-  }
-
-  const contextTypes = {
-    router: PropTypes.object,
+  const extendedOptions = {
+    ...defaultOptions,
+    ...options,
   }
 
-  const extendedOptions = Object.assign({}, defaultOptions, options, {
-    timeout: 100,
-  })
-
   const uniqueID = createUniqueIDFactory(extendedOptions.id)
   const uniqueIndex = createUniqueIndexFactory(1000)
 
-  class PortalWrapper extends Component {
-    constructor(props) {
-      super()
+  class PortalWrapper extends React.PureComponent {
+    static propTypes = propTypes
+    static defaultProps = {
+      isOpen: false,
+    }
+    static contextTypes = {
+      router: noop,
+    }
+    static childContextTypes = {
+      closePortal: noop,
+    }
+
+    static displayName = `withPortal(${getComponentName(ComposedComponent)})`
+
+    triggerComponent = null
+    triggerNode = null
+    _isMounted = false
+    _portalWrapperId = uniqueIndex()
+
+    // Welcome aboard, Mr. Manager!
+    _MrManager = setupManager(managerNamespace)
+    // Wow, I'm Mr. Manager!
+    // Well, manager… we we just say manager.
```

Comment:
🤣

Reply:
😂 !! I forgot about this until today

![](https://external-preview.redd.it/aFgX1ssVoPFPF46AysRTgnRaFxg4HUoEGq206xExT9c.jpg?width=500&auto=webp&s=8b080e6a9194b9d42e892159a50bc74b390869b1)